src/Compilers/CSharp/Portable/Symbols/ObsoleteAttributeHelpers.cs

```diff
@@ -79,10 +126,10 @@ internal static ThreeState GetObsoleteContextState(Symbol symbol, bool forceComp
         internal static DiagnosticInfo CreateObsoleteDiagnostic(Symbol symbol, BinderFlags location)
         {
             var data = symbol.ObsoleteAttributeData;
+            Debug.Assert(data != null);
```

Comment:
We could `throw ExceptionUtilities.Unreachable;` instead of assert and check, but since this method is only reporting warnings, it seemed like we could silently continue in this case.